### PR TITLE
[INLONG-403][Sort] Add doc about how to extend Extract or Load node

### LINKED ISSUE #403: [Sort] Add doc about how to extend Extract or Load node
Add doc about how to extend Extract or Load node

### PR DESCRIPTION
**Pull request by Oneal65**
Fixes #403

### Motivation

Add doc about how to extend Extract or Load node for new sort

### Modifications

remove how_to_write_plugin_sort_backup_ch.md
add how_to_extend_extract_or_load_node.md

### Verifying this change

- [x] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

**Review by EMsnap — COMMENTED** [review]
sort_uml is not supported to be in this commit

**Comment by Oneal65** [comment]
> sort_uml is not supported to be in this commit
thispicture has been cited in the md

File: docs/design_and_concept/how_to_extend_extract_or_load_node_en.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+
+**LoadNode**: destination abstraction for data synchronization
+
+**NodeRelationShip**:  abstraction of each node relationship in data synchronization
+
+**FieldRelationShip**:  abstraction of the relationship between upstream and downstream node fields in data synchronization
+
+**FieldInfo**: node field
+
+**BuiltInFieldInfo**: node built-in fields
```
Comment (gong) [inline docs/design_and_concept/how_to_extend_extract_or_load_node_en.md:38]: It should be `MetaFieldInfo` now

Comment (Oneal65) [reply docs/design_and_concept/how_to_extend_extract_or_load_node_en.md:38]: OK

File: docs/design_and_concept/how_to_extend_extract_or_load_node_en.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+# Overview
+
+InLong-Sort is known as a real-time ETL system.  Currently, supported extract or load includes elasticsearch, HBase, hive, iceberg, JDBC, Kafka, mongodb, mysql, orcale, Postgres, pulsar, etc。InLong-Sort is an ETL solution based on Flink SQL，The powerful expressive power of Flink SQL brings high scalability and flexibility. Basically, the semantics supported by Flink SQL are supported by InLong-Sort。In some scenarios, when the built-in functions of Flink SQL do not meet the requirements, they can also be extended through various UDFs in InLong-Sort. At the same time, it will be easier for those who have used SQL, especially Flink SQL, to get started.
+
+This article describes how to extend a new source (abstracted as extract node in inlong) or a new sink (abstracted as load node in inlong) in InLong-Sort.  After understanding the InLong-Sort architecture, you can understand how the source corresponds to the extract node, and how the sink corresponds to the load node. The architecture of inlong sort can be represented by UML object relation diagram as: 
+
+![sort_UML](img/sort_uml.png)
+
+The concepts of each component are:
```
Comment (dockerzhang) [inline docs/design_and_concept/how_to_extend_extract_or_load_node_en.md:14]: it's better to add a table to show the following item information.

File: docs/design_and_concept/how_to_extend_extract_or_load_node_en.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+**MetaFieldInfo**: node meta fields
+
+**Function**: abstraction of transformation function
+
+**FunctionParam**: input parameter abstraction of function
+
+**ConstantParam**: constant parameters
+
+To extend the extract node or load node, you need to do the following: 1.  Inherit the node class (such as MyExtractNode) and build specific extract or load usage logic; 2.  In a specific node class (such as MyExtractNode), specify the corresponding Flink connector; 3.  Use specific node classes in specific ETL implementation logic (such as MyExtractNode)
```
Comment (dockerzhang) [inline docs/design_and_concept/how_to_extend_extract_or_load_node_en.md:46]: To extend the extract node or load node, you need to do the following: 
- Inherit the node class (such as MyExtractNode) and build specific extract or load usage logic
- In a specific node class (such as MyExtractNode), specify the corresponding Flink connector
- Use specific node classes in specific ETL implementation logic (such as MyExtractNode)

File: docs/design_and_concept/how_to_extend_extract_or_load_node_en.md (hunk 2 above)
Comment (Oneal65) [reply docs/design_and_concept/how_to_extend_extract_or_load_node_en.md:14]: DONE

File: docs/design_and_concept/how_to_extend_extract_or_load_node_en.md (hunk 3 above)
Comment (Oneal65) [reply docs/design_and_concept/how_to_extend_extract_or_load_node_en.md]: DONE

**Review by dockerzhang — APPROVED** [review]

File: docs/design_and_concept/how_to_extend_extract_or_load_node_en.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
+---
+title: Sort Plugin
+sidebar_position: 3
+---
+
+# Overview
```
Comment (yunqingmoswu) [inline docs/design_and_concept/how_to_extend_extract_or_load_node_en.md:6]: `#` -> `##`

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+...
+@JsonSubTypes({
+        @JsonSubTypes.Type(value = KafkaLoadNode.class, name = "kafkaLoad")
+})
+public interface Node {...}
+```
+
+**第三步**：扩展flink connector ，Kafka的sort connector在incubator-inlong/inlong-sort/sort-connectors/kafka
+
+# 集成Extract 和Load 到InLong-Sort主流程
```
Comment (yunqingmoswu) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md]: 英文单词前后空格

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+public interface Node {...}
+```
+
+**第三步**：扩展flink connector，查看该（/incubator-inlong/inlong-sort/sort-connectors/mongodb-cdc）目录下是否已经存在对应的connector。如果还没有，则需要参考flink官方文档[DataStream Connectors ](https://nightlies.apache.org/flink/flink-docs-release-1.13/docs/connectors/datastream/overview/#datastream-connectors)来扩展，直接调用已有的flink-connector（例如incubator-inlong/inlong-sort/sort-connectors/mongodb-cdc）或自行实现相关的connecter。
+
+# 扩展Load Node
+
+扩展一个LoadNode分为三步骤：
+
+**第一步**：继承LoadNode类，类的位置在incubator-inlong/inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/LoadNode.java；在实现的LoadNode中指定connecter；
```
Comment (yunqingmoswu) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md:98]: incubator-inlong/inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/LoadNode.java -> `incubator-inlong/inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/LoadNode.java`

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md [hunk 7]
```diff
@@ -0,0 +1,205 @@
+---
+title: Sort 插件
+sidebar_position: 3
+---
+
+# 总览
+
+InLong-Sort是一个ETL系统，当前支持的extract或load包括elasticsearch、hbase、hive、iceberg、jdbc、kafka、mongodb、mysql、orcale、postgres、pulsar等。InLong-Sort是基于Flink SQL的ETL方案，Flink SQL强大的表达能力带来的高可扩展性、灵活性，基本上Flink SQL支持的语意，InLong-Sort都支持。在个别场景，Flink SQL内置的函数不满足需求时，还可通过各种UDF来扩展。同时对于曾经使用过SQL尤其是使用过Flink SQL的人而言，会更容易上手。
```
Comment (yunqingmoswu) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md:8]: elasticsearch、hbase、hive、iceberg、jdbc、kafka、mongodb、mysql、orcale、postgres、pulsar这些首字符大写，前后空格

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+| FunctionParam     | 函数的入参抽象                                              |
+| ConstantParam     | 常量参数                                                    |
+
+扩展Extract Node 或 Load Node需要做的工作是：
+
+- 继承Node类（例如MyExtractNode），构建具体的extract 或 load使用逻辑；
+- 在具体的Node类（例如MyExtractNode）中，指定对应Flink connector；
+- 在具体的ETL实现逻辑中使用具体的Node类（例如MyExtractNode）。
+
+其中第二步中可以使用已有的flink connector，或者用户自己扩展，如何扩展flink connector请参考flink官方文档[DataStream Connectors ](https://nightlies.apache.org/flink/flink-docs-release-1.13/docs/connectors/datastream/overview/#datastream-connectors).
```
Comment (yunqingmoswu) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md:40]: flink connector  -> Flink Connectors

Comment (yunqingmoswu) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md:40]: flink -> Flink

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md [hunk 9]
```diff
@@ -0,0 +1,205 @@
...
+
+# 扩展Extract Node
+
+扩展一个ExtractNode分为三步骤：
+
+**第一步**：继承ExtractNode类，类的位置在incubator-inlong/inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/ExtractNode.java；在实现的ExtractNode中指定connecter；
```
Comment (yunqingmoswu) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md:46]: incubator-inlong/inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/ExtractNode.java -> `incubator-inlong/inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/ExtractNode.java`

File: docs/design_and_concept/how_to_extend_extract_or_load_node_en.md (hunk 4 above)
Comment (Oneal65) [reply docs/design_and_concept/how_to_extend_extract_or_load_node_en.md]: DONE

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md [hunk 10]
```diff
@@ -0,0 +1,205 @@
...
+public abstract class ExtractNode implements Node{...}
+
+...
+@JsonSubTypes({
+        @JsonSubTypes.Type(value = MongoExtractNode.class, name = "mongoExtract")
+})
+public interface Node {...}
+```
+
+**第三步**：扩展flink connector，查看该（/incubator-inlong/inlong-sort/sort-connectors/mongodb-cdc）目录下是否已经存在对应的connector。如果还没有，则需要参考flink官方文档[DataStream Connectors ](https://nightlies.apache.org/flink/flink-docs-release-1.13/docs/connectors/datastream/overview/#datastream-connectors)来扩展，直接调用已有的flink-connector（例如incubator-inlong/inlong-sort/sort-connectors/mongodb-cdc）或自行实现相关的connecter。
```
Comment (yunqingmoswu) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md:92]: flink-connector -> Sort Connectors

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md [hunk 11]
```diff
@@ -0,0 +1,205 @@
...
+        @JsonSubTypes.Type(value = KafkaLoadNode.class, name = "kafkaLoad")
+})
+public interface Node {...}
+```
+
+**第三步**：扩展flink connector ，Kafka的sort connector在incubator-inlong/inlong-sort/sort-connectors/kafka
+
+# 集成Extract 和Load 到InLong-Sort主流程
+
+将Extract和Load集成到InLong-Sort主流程中，需要构建总览小节中提到的语意：Group、stream、node等。InLong-Sort的入口类在inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/Entrance.java。Extract和Load如何集成至InLong-Sort，可参考下面的UT，首先构建对应的ExtractNode、LoadNode，再构建NodeRelation、streamInfo、groupInfo，最后使用FlinkSqlParser去执行。
```
Comment (yunqingmoswu) [inline i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md:161]: inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/Entrance.java -> `inlong-sort/sort-core/src/main/java/org/apache/inlong/sort/Entrance.java`

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md (hunk 8 above)
Comment (Oneal65) [reply i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md:40]: DONE

File: docs/design_and_concept/how_to_extend_extract_or_load_node_en.md [hunk 12]
```diff
@@ -0,0 +1,208 @@
+---
+title: Sort Plugin
+sidebar_position: 3
+---
+
+## Overview
+
+InLong-Sort is known as a real-time ETL system.  Currently, supported extract or load includes elasticsearch, HBase, hive, iceberg, JDBC, Kafka, mongodb, mysql, orcale, Postgres, pulsar, etc。InLong-Sort is an ETL solution based on Flink SQL，The powerful expressive power of Flink SQL brings high scalability and flexibility. Basically, the semantics supported by Flink SQL are supported by InLong-Sort。In some scenarios, when the built-in functions of Flink SQL do not meet the requirements, they can also be extended through various UDFs in InLong-Sort. At the same time, it will be easier for those who have used SQL, especially Flink SQL, to get started.
```
Comment (healchow) [inline docs/design_and_concept/how_to_extend_extract_or_load_node_en.md:8]: It is recommended to use the official name for those extracts or load nodes.

File: i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md (hunk 10 above)
Comment (Oneal65) [reply i18n/zh-CN/docusaurus-plugin-content-docs/current/design_and_concept/how_to_extend_extract_or_load_node_ch.md:92]: DONE

File: docs/design_and_concept/how_to_extend_extract_or_load_node_en.md (hunk 12 above)
Comment (Oneal65) [reply docs/design_and_concept/how_to_extend_extract_or_load_node_en.md:8]: DONE

**Review by healchow — APPROVED** [review]

**Pull request merged**